lib/modules/datasource/glasskube-packages/index.ts

```diff
@@ -0,0 +1,76 @@
+import { cache } from '../../../util/cache/package/decorator';
+import { joinUrlParts } from '../../../util/url';
+import * as glasskubeVersioning from '../../versioning/glasskube';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import {
+  GlasskubePackageManifestYaml,
+  GlasskubePackageVersions,
+  GlasskubePackageVersionsYaml,
+} from './schema';
+
+export class GlasskubePackagesDatasource extends Datasource {
+  static readonly id = 'glasskube-packages';
+  static readonly defaultRegistryUrl =
+    'https://packages.dl.glasskube.dev/packages';
+  override readonly customRegistrySupport = true;
+  override defaultVersioning = glasskubeVersioning.id;
+
+  override defaultRegistryUrls = [
+    GlasskubePackagesDatasource.defaultRegistryUrl,
+  ];
+
+  constructor() {
+    super(GlasskubePackagesDatasource.id);
+  }
+
+  @cache({
+    namespace: `datasource-${GlasskubePackagesDatasource.id}`,
+    key: ({ registryUrl, packageName }: GetReleasesConfig) =>
+      [registryUrl, packageName].filter((s) => s).join('-'),
```

Comment:
Why the noop filter with a join?

```suggestion
      `${registryUrl}-${packageName}`
```

Reply:
Because `registryUrl` can be undefined, at least according to the type definition (which would result in `"-foo"`). But if that doesn't matter, your proposed change makes sense.

Reply:
`registryUrl` can not be `undefined` as long there is a `defaultRegistryUrl` defined.